plugins/debuginfo-install.py

```diff
@@ -79,54 +83,95 @@ def run(self):
         errors_spec = []
 
         for pkgspec in self.opts.package:
-            package_query = sorted(dnf.subject.Subject(pkgspec).get_best_query(
-                self.cli.base.sack).filter(arch__neq='src'), reverse=True)
-            if not package_query:
-                msg = _('No match for argument: %s')
-                logger.info(msg, self.base.output.term.bold(pkgspec))
+            solution = dnf.subject.Subject(pkgspec).get_best_solution(self.base.sack)
+
+            query = solution["query"].filterm(arch__neq='src')
+
+            packages = query.available()._na_dict()
+            # installed versions of packages have priority, replace / add them to the dict
+            packages.update(query.installed()._na_dict())
+
+            # Remove debuginfo packages if their base packages are in the query.
+            # They can get there through globs and they break the installation
+            # of debug packages with the same version as the installed base
+            # packages. If the base package of a debuginfo package is not in
+            # the query, the user specified a debug package on the command
+            # line. We don't want to ignore those, so we will install them.
+            # But, in this case the version will not be matched to the
+            # installed version of the base package, as that would require
+            # another query and is further complicated if the user specifies a
+            # version themselves etc.
+            for na in list(packages.keys()):
+                if na[0].endswith(Package.DEBUGINFO_SUFFIX):
+                    if (na[0][:-len(Package.DEBUGINFO_SUFFIX)], na[1]) in packages:
+                        packages.pop(na)
+                if na[0].endswith(Package.DEBUGSOURCE_SUFFIX):
+                    if (na[0][:-len(Package.DEBUGSOURCE_SUFFIX)], na[1]) in packages:
+                        packages.pop(na)
+
+            if not packages:
```

Comment:
The if could be moved several lines up and could end with "continue" line, because there is no packages.key() and no package.values(). Pop up statement still keep non-debuginfo packages in the dict.